websites/Q/Qobuz/presence.ts

```diff
@@ -0,0 +1,120 @@
+const presence = new Presence({
+  clientId: "921861694190407730"
+});
+
+async function getStrings() {
+  return presence.getStrings(
+    {
+      play: "general.playing",
+      pause: "general.paused",
+      viewAlbum: "general.buttonViewAlbum",
+      viewPlaylist: "general.buttonViewPlaylist"
+    },
+    await presence.getSetting("lang").catch(() => "en")
+  );
+}
+
+(async () => {
+  let strings = await getStrings(),
+    oldLang: string = null;
+
+  presence.on("UpdateData", async () => {
+    if (!document.querySelector("#root"))
+      return presence.setActivity({ largeImageKey: "logo" });
+
+    const [newLang, timestamps, cover] = await Promise.all([
+      presence.getSetting("lang").catch(() => "en"),
+      presence.getSetting("timestamps"),
+      presence.getSetting("cover")
+    ]);
+
+    oldLang ??= newLang;
+    if (oldLang !== newLang) {
+      oldLang = newLang;
+      strings = await getStrings();
+    }
+
+    const presenceData: PresenceData = {
+        largeImageKey: cover
+          ? document
+              .querySelector<HTMLImageElement>(
+                'div[class="player__track-cover"] img'
+              )
+              .src.replaceAll("230", "600")
+          : "logo"
+      },
+      songTitle = document.querySelector<HTMLAnchorElement>(
+        'a[class="player__track-name"]'
+      ),
+      fromPlaylist = !!document.querySelectorAll(
+        'div[class="player__track-album"] a'
+      )[2],
+      currentTime = document
+        .querySelector<HTMLElement>('span[class="player__track-time-text"]')
+        .innerText.split(":"),
+      endTime = document
+        .querySelectorAll<HTMLElement>(
+          'span[class="player__track-time-text"]'
+        )[1]
+        .innerText.split(":"),
+      currentTimeSec =
+        (parseFloat(currentTime[0]) * 60 + parseFloat(currentTime[1])) * 1000,
+      endTimeSec =
+        (parseFloat(endTime[0]) * 60 + parseFloat(endTime[1]) + 1) * 1000,
```

Comment:
arent you able to use the presence.getFormatTimestamp for this? (I think its called something like that)

Reply:
I don't know about this...

Reply:
presence.timestampFromFormat

Reply:
thanks